common/client-libs/gateway-client/src/client.rs

```diff
@@ -379,21 +379,26 @@ impl GatewayClient {
                 ws_stream,
                 self.local_identity.as_ref(),
                 self.gateway_identity,
-                self.coconut_credential.clone(),
             )
             .await
             .map_err(GatewayClientError::RegistrationFailure),
             _ => unreachable!(),
         }?;
-
-        self.authenticated = true;
-        Ok(shared_key)
+        self.authenticated = match self.read_control_response().await? {
+            ServerResponse::Register { status } => Ok(status),
+            ServerResponse::Error { message } => Err(GatewayClientError::GatewayError(message)),
+            _ => Err(GatewayClientError::NotAuthenticated),
+        }?;
+        if self.authenticated {
+            self.shared_key = Some(Arc::new(shared_key));
+        }
+        Ok(())
     }
 
-    pub async fn authenticate(
+    async fn authenticate(
```

Comment:
If authenticate is no longer returning authentication status then... what's the point of it?

Reply:
This is just a change to have `register` and `authenticate` functions more homogeneous. This makes them both do the changes on the mutable `self`, rather than one doing some changes in the function, and the other in the function caller, based on the returned value.